src/monster.cpp

```diff
@@ -146,54 +146,41 @@ void monster::poly(mtype *t)
 
 bool monster::can_upgrade() const
 {
-    // If we don't upgrade
+    // Turned off means turned off
+    if (!ACTIVE_WORLD_OPTIONS["MONSTER_UPGRADE_FACTOR"]) {
```

Comment:
Profile and find out, if it's used enough it can be promoted to one of the special option globals that are set when the option is set, then checking it is nearly free.